src/AspNet.Security.OAuth.Weixin/WeixinAuthenticationHandler.cs

```diff
@@ -109,13 +126,20 @@ protected override async Task<OAuthTokenResponse> ExchangeCodeAsync(string code,
 
         protected override string BuildChallengeUrl(AuthenticationProperties properties, string redirectUri)
         {
+            var oauthstate = Options.StateDataFormat.Protect(properties);
+            if (Options.AuthorizationEndpoint != WeixinAuthenticationDefaults.AuthorizationEndpoint)
```

Comment:
Use `Use string.Equals(a, b, StringComparison.Ordinal)`. In fact, as it's being done twice, it could be factored out into a method:

```csharp
private bool IsWeixinAuthorizationEndpointInUse()
{
    return string.Equals(Options.AuthorizationEndpoint, WeixinAuthenticationDefaults.AuthorizationEndpoint, StringComparison.Ordinal);
}
```